### PR TITLE
Content: Fighter Pod Utilization

### PR DESCRIPTION
**Pull request by Zitchas**
**Content (Outfitting Ships)**

## Summary
This PR implements the missile pods added in #5793 on stock & variant fighter, drone, interceptor, light transport, and light freighter ships. This is anticipated to be the first of several phases of changes for fighters. 

**Note:** 
This PR does not change the base hull statistics of any ship. This means that players (or NPCs in missions) are perfectly capable of stripping out the pods and putting in a launcher. For now, anyway. This phase has several goals:
A) Arm small combatants (fighters, drones, light transports, and light freighters.) as appropriate fighter type armaments. Meaning, as small ships that are intended to fly out, bombard a target, return to carrier to re-equip, repeat. 
B) Encourage fighter-carrier behavior that actually reflects the "short-term launched combatant" style that we expect from fighters.
C) Make early-game combat a little more reasonable. (Someone in a Clipper facing off against another clipper should be a reasonable fight, not a "face down enough firepower to seriously damage a falcon" situation. ) Lone small ship vs small ship combat should give the player some chance to engage. Another bad example: One arrow taking on another arrow. Each arrow (prior to this PR) carries 45 sidewinders. I don't think there are any arrow builds out there that can survive that big of a barrage. And just to be clear, I do not consider "I can obliterate you at the same time you obliterate me" to be a reasonable fight.)

**Phases of fighter changes:**
Right now, I'm picturing a multi-phase approach. Note that none of these stages is mandatory, and they don't have to be done in this order. That being said, the main boost to fighter's utility is likely to be in phase 3, and whenever VitalChip's fighter AI gets merged in.
1) Redo fighter load-outs to be actually fighter-appropriate armaments. (this will, by its very nature, result in two things: a) fighters will require rearming at carriers to be long-term effective if they use projectiles, and b) fighters will deal less damage overall.
2) Re-do carriers so that they bring sufficient fighters/drones to a fight for them to actually be significant in battle. This will probably also require carriers outfits to be changed around to carry more spare ammo with which to restock fighters. In short, carriers become more like actual carriers and less like cruisers-that-have-air-support.
3) Adjust fighter mass & propulsion so that fighters actually have movement stats like we'd expect fighters to have. (as opposed to what they are right now, which is basically slightly-faster-than-light warships)
4) Adjust the size of fighter/drone sprites so they are a little bit smaller and thus a little harder to hit (My expectation is this will be minor change, like fighters & drones being reduced down to 75% of their current scale. A little smaller, not a lot smaller.

Somewhere amidst all this I hope to get @Vitalchip 's fighter AI improvements into the game too. Probably in or around step 3.

## PR Checklist
 - ~~[X] I updated the copyright attributions, or decline to claim copyright of any assets produced or modified~~
 - ~~[X] I uploaded the necessary image, blend, and texture assets here: {{insert link to assets}}~~
 - ~~[X] I created a PR to the [endless-sky-high-dpi repo](https://github.com/endless-sky/endless-sky-high-dpi) with the `@2x` versions of these art assets: {{insert PR link}}~~

edit: Some comments about ship purpose and outfitting:
In many cases I swapped it out for the pods in an effort to have more of a progression for new pilots. Due to the amount of missiles a standard launcher can hold (vastly more than any light ship has a hope of dealing with), interceptors starts (and for that matter, combat in interceptors and other small ships, typically against pirates) are practically impossible. 

So while yes, some light warships (particularly military ones as opposed to civilian ones) should carry full launchers, we still need to have at least *some* light warships with pods instead. It comes down to role, though.

Pods:
- Ships aimed at taking out single (usually light) ships
- Ships that are generally "strike and run"
- Ships that operate close to somewhere they can resupply (carrier or station)

Launchers:
- Ships aimed at taking out fleets and/or heavy warships
- Ships that generally "hold the line" and are expected to control a system for long periods of time and/or protracted battles
- Ships that operate at long ranges (aka multiple hyperjumps) from support and are expected to go through many battles before their next resupply.

Keep in mind that there's also a discrepancy in NPC vs Player outfitting.  The player is virtually *always* going to be falling into that last category "operating at long range from support and expecting to go through multiple battles before their next resupply," whereas NPCs don't. I know thematically, many NPCs *would* operate that way, but a) it's expensive and they have a budget, and b) it makes the game a lot harder for the player. (See: All the comments we get about new players getting annihilated by "small" pirate spawns.)

So, long story short, part of the purpose of pods is making it so that small ship vs small ship combat is actually a thing, and that will involve ensuring that some NPC small ships (like the headhunter) are outfitted as if they were tracking down someone in a small ship, not always rigging them for maximized "how much damage potential can we cram into this hull."

### CODE REVIEW

File: data/human/variants.txt [hunk 1]
```diff
@@ -1161,8 +1161,8 @@ ship "Fury" "Fury (Missile)"
 	outfits
 		"Javelin Pod"
 		"Javelin" 200
-		"Meteor Missile Launcher" 2
-		"Meteor Missile" 75
+		"Meteor Missile Pod" 2
+		"Meteor Missile" 14
 		"Meteor Missile Box"
```
Comment (tobynorth) [inline data/human/variants.txt:1166]: This missile box is now unused. Worth removing?

**Review by tobynorth — CHANGES_REQUESTED** [review]
Looks pretty good. I added some line comments, and here are a couple others I couldn't add as line comments:

- A few interceptor variants still have the standard Javelin Pod (rather than the Mini). These are: Wasp (Javelin), Fury (Heavy), Fury (Missile)
- Update missile descriptions as appropriate. For example, the Meteor Missile description refers to needing the Meteor Missile Launcher, as well as its 30 missile capacity. (Should probably just remove the latter part entirely... will be easy to forget to keep the capacity up to date in this desc if it changes again :) )

Also, a couple things I want to bring to your attention in case you hadn't thought of them yet, but which would probably be better suited for later phases of fighter changes:

- Consider creating pod versions of relevant non-human secondary weapons. Looks like the Hai Tracker Pod might be the only one, in practice
- Consider slightly reducing affected ship costs to reflect the smaller capacities

File: data/human/ships.txt [hunk 2]
```diff
@@ -96,8 +96,8 @@ ship "Argosy"
 			"hit force" 900
 	outfits
 		"Energy Blaster" 2
-		"Meteor Missile Launcher" 2
-		"Meteor Missile" 60
+		"Meteor Missile Pod" 2
```
[the remark this answers is not in the document]
Comment (Zitchas) [reply data/human/ships.txt:99]: Fair enough. I'll revert it.

File: data/human/ships.txt [hunk 3]
```diff
@@ -155,8 +155,8 @@ ship "Arrow"
 			"hull damage" 120
 			"hit force" 360
 	outfits
-		"Sidewinder Missile Launcher"
-		"Sidewinder Missile" 45
+		"Sidewinder Missile Pod" 2
```
[the remark this answers is not in the document]
Comment (Zitchas) [reply data/human/ships.txt:158]: Yes. It struck me that the arrow, being a VIP transport, might make sense to be able to do a bit more than a stock fighter (the arrow has nowhere to restock, whereas a fighter by definition has either a carrier or a home base to restock from)

And the Hai variant also has two pods.

File: data/human/variants.txt [hunk 4]
```diff
@@ -1368,8 +1368,8 @@ ship "Headhunter" "Headhunter (Strike)"
 	outfits
 		"Javelin Pod" 2
 		"Javelin" 400
-		"Meteor Missile Launcher" 2
-		"Meteor Missile" 60
+		"Meteor Missile Pod" 2
```
[the remark this answers is not in the document]
Comment (Zitchas) [reply data/human/variants.txt:1371]: It is, however I swapped it out for the pods in an effort to have more of a progression for new pilots. Due to the amount of missiles a standard launcher can hold (vastly more than any light ship has a hope of dealing with), interceptors starts (and for that matter, combat in interceptors and other small ships, typically against pirates) are practically impossible. 

So while yes, some light warships (particularly military ones as opposed to civilian ones) should carry full launchers, we still need to have at least *some* of them with pods instead. It comes down to role, though.

Pods:
- Ships aimed at taking out single (usually light) ships
- Ships that are generally "strike and run"
- Ships that operate close to somewhere they can resupply (carrier or station)

Launchers:
- Ships aimed at taking out fleets and/or heavy warships
- Ships that generally "hold the line" and are expected to control a system for long periods of time and/or protracted battles
- Ships that operate at long ranges (aka multiple hyperjumps) from support and are expected to go through many battles before their next resupply.

Keep in mind that there's also a discrepancy in NPC vs Player outfitting.  The player is virtually *always* going to be falling into that last category "operating at long range from support and expecting to go through multiple battles before their next resupply," whereas NPCs don't. I know thematically, many NPCs *would* operate that way, but a) it's expensive and they have a budget, and b) it makes the game a lot harder for the player. (See: All the comments we get about new players getting annihilated by "small" pirate spawns.)

File: data/human/variants.txt [hunk 5]
```diff
@@ -339,10 +339,10 @@ ship "Berserker" "Berserker (Afterburner)"
 
 ship "Berserker" "Berserker (Strike)"
 	outfits
-		"Heavy Rocket Launcher"
-		"Heavy Rocket" 20
-		"Sidewinder Missile Launcher"
-		"Sidewinder Missile" 45
+		"Heavy Rocket Pod"
+		"Heavy Rocket" 4
```
[the remark this answers is not in the document]
Comment (Zitchas) [reply data/human/variants.txt:343]: Thank you, fixed. (for all three.)

File: data/human/variants.txt (hunk 1 above)
Comment (Zitchas) [reply data/human/variants.txt:1166]: Yes, removed. Thanks!

File: data/human/variants.txt [hunk 6]
```diff
@@ -1269,10 +1269,10 @@ ship "Hauler III" "Hauler III (Hai)"
...
 ship "Hawk" "Hawk (Bomber)"
 	outfits
-		"Heavy Rocket Launcher"
-		"Heavy Rocket" 20
-		"Torpedo Launcher"
-		"Torpedo" 30
+		"Heavy Rocket Pod"
+		"Heavy Rocket" 4
+		"Torpedo Pod"
+		"Torpedo" 6
```
[the remark this answers is not in the document]
Comment (Zitchas) [reply data/human/variants.txt:1275]: Fixed. Thanks!

File: data/human/ships.txt (hunk 3 above)
[the remark this answers is not in the document]
Comment (tobynorth) [reply data/human/ships.txt:158]: Makes sense about adding the extra pod. I just brought up the Hai variant because it already had two pods prior to your changes, so I wasn't sure if we would want to maintain its (slight) armament advantage over the stock version by bumping it to 3 or 4 pods. Definitely not a big deal either way.

File: data/human/variants.txt (hunk 4 above)
[the remark this answers is not in the document]
Comment (tobynorth) [reply data/human/variants.txt:1370]: Cool, sounds good to me. I'm not too familiar yet with the roles of different ships beyond the official categories in the ships data file, so I will definitely take your word for it.

As a side note, your last paragraph makes me think it might be neat to make it so some NPCs spawn with a quantity of secondary weapon ammo that is randomized in some way -- perhaps dependent on how professional vs. ragtag the fleet in question is supposed to be, and how close to combat zones or far from "home" they are. Seems like it would improve the game thematically like you mentioned, without necessarily making the game harder. Obviously out of scope for this PR, though.

**Comment by tobynorth** [comment]
Your latest changes all look good! Are you intentionally skipping [these suggestions](https://github.com/endless-sky/endless-sky/pull/6203#pullrequestreview-736703022) other than the weapon desc updates, though? I don't see changes related to those. The interceptor variants that still have standard javelin pods seem particularly worth updating to me.

**Comment by tobynorth** [comment]
Oh and also, don't forget to update weapon descs for torpedoes, typhoon torpedoes, and heavy rockets, too.

**Review by WarlordMike — APPROVED** [review]
Looks like a good road map. And the changes proposed make a lot of sense. It'll also help keep new players from getting nuked so quickly, which I'm 100% supportive of.

**Comment by Pointedstick** [comment]
This PR highlights my initial concern: replacing launchers with pods 1-for-1 (or even several for one) is a straight-up downgrade for most ships. If a ship can fit a launcher as well as everything else they still need, then there is no reason to use a pod for the same missile instead. This is especially true of the small launchers for Meteors and Javelins. I think in this case we need to upgrade: replace one meteor missile launcher with one rocket or torpedo pod, for example.

However stepping back, if we are going to put pods on fighters, I think we also need to expand the number of available gun ports to make this change really work. A 6-gun fighter would be able to make use of pods much better: for example two guns, 2 rocket/torpedo pods, and 2 meteor/sidewinder pods. Or 4 guns and 2 rocket/torpedo pods.

Absent that, I think we need to upgrade the ordnance so that one launcher of a weak missile is replaced with one pod for a stronger missile. Otherwise we are just downgrading these ships' combat capabilities with nothing to offset it.

**Comment by Zitchas** [comment]
> This PR highlights my initial concern: replacing launchers with pods 1-for-1 (or even several for one) is a straight-up downgrade for most ships. If a ship can fit a launcher as well as everything else they still need, then there is no reason to use a pod for the same missile instead. This is especially true of the small launchers for Meteors and Javelins. I think in this case we need to upgrade: replace one meteor missile launcher with one rocket or torpedo pod, for example.

Well, this is kind of the fundemental problem with balancing stuff in ES: Everything is equipped as if it expects to be going toe to toe with a heavy warship tomorrow, which means that combat in anything but a heavy warship is terrible because (baring alien AM) everything dies to missile-fire-calibrated-to-kill-heavy-warships. This is why new pilots die so often if they leave safe systems, try to take on something their own size, or acquire significant amounts of cargo space.

As I've mentioned from the beginning, this is just one step in a longer process. First step was to give ourselves some options. Second step is to put those options in use. Yes, this is flat up a nerf for the fighters involved. because we are pulling fighters down from "large warship in every respect except hp" and sticking them back in the "small ships that harass" category. The problem that this is trying to solve is to make small ship combat *possible* for the people who most often fly small ships: New players. 

> 
> However stepping back, if we are going to put pods on fighters, I think we also need to expand the number of available gun ports to make this change really work. A 6-gun fighter would be able to make use of pods much better: for example two guns, 2 rocket/torpedo pods, and 2 meteor/sidewinder pods. Or 4 guns and 2 rocket/torpedo pods.

Yes, I fully agree with this. But given the nature of development and everyone liking to take things one slow step at a time, the first step is to get them into the right caliber of weaponry, *then* buff them so they can be outfitted to blaster spam or whatever the case may be.

> 
> Absent that, I think we need to upgrade the ordnance so that one launcher of a weak missile is replaced with one pod for a stronger missile. Otherwise we are just downgrading these ships' combat capabilities with nothing to offset it.

I fully expect that one of the later steps (maybe the next step) will involve bumping up all the carriers to carry more drones & fighters than they currently do, as I think we can all agree that the point of carriers isn't to launch one or two fighters that devastate the enemy, but rather to launch a flight or even a swarm of fighters that collectively devastate the enemy. And that may well include some bombers with heavy rockets and stuff. But for now I think we need to stay away from having "punching above their weight" loadouts as stock. Quite possibly *double* or even triple the number of bays across the board.

**Comment by Pointedstick** [comment]
Yeah OK, fair enough.

**Comment by Pointedstick** [comment]
To be clear, my position remains that carried interceptors should at least have somewhat of a plausible purpose for existing based on their inherent characteristics.

In the real world, the only reason for carried aircraft to exist is to increase the threat range of the launching platform; a fighter carrying a gun or a missile gives the launching platform a greater threat range compared to if it carried that gun or missile itself, because the weapon's own threat range is added to the fighter's combat range . However, if the launching platform had a threat range equal to fighter + armament, or if the increased threat range was not useful in practical combat situations, than the launching platform would be much better off stocking more weapons on its own. They would take up less space, be cheaper, and not put humans in harm's way. Launching facilities are *big*.

The fundamental problem in Endless Sky is that this increased threat range of interceptors is generally not relevant in fights, for a couple reasons:
1. Ships usually spawn very close to one another; range often doesn't even enter into the equation at all
2. The AI makes ships always want to close to knife fighting distance unless they have zero guns; even one gun changes the AI to "shoot you in the face with my guns!" mode
3. Capital ships are really durable; the real-world reason why you want your carrier to stay far away is that it's quite vulnerable relative to its cost; one good torpedo, missile, or bomb hit could cripple or sink it. ES capital ships, on the other hand, are very tanky. They can take a lot of abuse and don't suffer from diminished capabilities as they take more damage.

All of this leads logically to a dynamic where tanky capital ships close to knife fighting distance and pummel each other with huge guns, while small vulnerable fighters and interceptors that cannot take any hits need to spam missiles from extreme distance to have any chance of surviving. It's logical, but I don't think it's very fun to play, and it kind of violates the expectations we have from other Sci-Fi games and movies that more or less mimic a World War II naval battle in space.

**Comment by Zitchas** [comment]
I fully agree with that. The way things are now, fighters don't serve a whole lot of purpose. But at least now they can serve as something for early pilots to learn to fight against. That being said, we're moving forward. 

List of things that I'm working towards getting in the game to make fighters useful. Not all will be needed. But many of them.
- More fighters per carrier (at least on dedicated carrier ships. a couple fighters is fine on a cruiser-that-happens-to-have-a-bay. But a dedicated carrier? True carriers should be looking at a minimum of double the fighters they currently carry.)
- Fighters that actually have brains ( @vitalchip 's AI improvements for fast ship manouvering)
- Fighters that are actually *zippy* compared to even the fastest warships (by zippy, I mean "can launch from their carrier, approach their target, launch munitions, and return to the carrier to reload **before** their carrier and the target heavy warship have been able to close to knife fight range.")
- Hyperjump arrival range set to at *least* +1000 from what it is now, possibly even +2000.
- More gunports per fighter
- Single-munition-per-hardpoint capabilities.

**Comment by Pointedstick** [comment]
Sounds like a good list to me. :+1:
